src/handlers.rs

```diff
@@ -59,30 +55,23 @@ impl<S> FromRequest<S> for HawkHeader {
 }
 
 macro_rules! endpoint {
-    ($handler:ident: $dispatcher:ident ($path:ident: $path_type:ty $(, $param:ident: $type:ty)*) {$($property:ident: $value:expr),*}) => {
+    ($handler:ident: $data:ident ($path:ident: $path_type:ty $(, $param:ident: $type:ty)*) {$($property:ident: $value:expr,)*}) => {
         pub fn $handler(
-            ($path, state$(, $param)*): (Path<$path_type>, State<ServerState>$(, $type)*),
+            ($path, _state$(, $param)*): (Path<$path_type>, State<ServerState>$(, $type)*),
         ) -> FutureResponse<HttpResponse> {
-            state
-                .db_executor
-                .send(dispatcher::$dispatcher {
-                    $($property: $value),*
-                })
-                .from_err()
-                .and_then(|res| match res {
-                    Ok(info) => Ok(HttpResponse::Ok().json(info)),
-                    Err(_) => Ok(HttpResponse::InternalServerError().into()),
-                })
-                .responder()
+            let _data = data::$data {
+                $($property: $value,)*
+            };
+            Box::new(result(Ok(HttpResponse::Ok().json(()))))
         }
     }
 }
 
 macro_rules! info_endpoints {
-    ($($handler:ident: $dispatcher:ident),+) => ($(
+    ($($handler:ident: $data:ident,)+) => ($(
```

Comment:
I tweaked the match arms for all these macros so that trailing items keep their comma. It makes our macro invocations better match the real Rust code (not to mention all the other reasons that trailing commas are a Good Thing).